src/utils/intrusive_ptr.h

```diff
@@ -41,6 +41,10 @@ class intrusive_ptr {
     this->Swap(other);
     return *this;
   }
+  
+  bool operator==(intrusive_ptr<T> other) {
+    return this->ptr_ == other.ptr_;
+  }
```

Comment:
`operator==` should be declared as taking a `const` pointer to `this`. This is done by putting a `const` modifier trailing the signature. Making no other changes, this would be written like so:

```C++
bool operator==(intrusive_ptr<T> other) const {
```

In addition, the use of the parameter as a value type rather than a const reference type introduces a copy. That would be fine (indeed, preferable!) for many types, but `intrusive_ptr` does a pointer dereference and modification on every copy, which makes this potentially have surprising effects. I'd say to change the parameter type to

```C++
const intrusive_ptr<T> &other
````

which is the more standard form for operator-overloading parameters anyway.

Reply:
That makes sense. I will update the code.